stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/logic/internal/LogicRegistry.kt

```diff
@@ -122,14 +122,14 @@ internal class LogicRegistry internal constructor(
          * @param repos [RepositoryFacade] to interact with local data sources.
          * @param client An instance of [ChatClient].
          */
-        internal fun getOrCreate(
+        internal fun create(
```

Comment:
Should we clear the existing instance when the user disconnects? 🤔

Reply:
So the solution can be clearing the instance when user disconnects and use old `getOrCreate` approach. Just not sure if in case of any race condition, the previous `instance` is not cleared and we try to call `getOrCreate`, it can return old instance.

Reply:
To be safe, we can clear the instance and as well as use `create` method. Also throw exception in `create` method if the instance was not null.

Reply:
I'm not sure if we should throw an exception as we want to avoid crashing - I would rather add a log for now